compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -3112,12 +3112,22 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
             drop_span, borrow_span
         );
 
+        // `TerminatorKind::Return`'s span (the `drop_span` here) `lo` can be subtly wrong and point
+        // at a single character after the end of the function. This is somehow relied upon in
+        // existing diagnostics, and changing this in `rustc_mir_build` makes diagnostics worse in
```

Comment:
I can't remember, was this used to make sure the span is never zero-width?

Reply:
I don't really know if the hi == low is that zero-width. The hi could be seen as exclusive, so it can make sense that it's pointing after the end of function, but lo being also there is quite strange.